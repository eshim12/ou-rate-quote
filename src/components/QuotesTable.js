import React from 'react'
import Spinner from 'react-bootstrap/Spinner'
import Table from 'react-bootstrap/Table'

const QuotesTable = ({quotes, loading}) => (
	<div>
		<Table bordered>
			<thead>
				<tr>
					<th>Lender</th>
					<th>Product</th>
					<th>Rate</th>
					<th>Closing Costs</th>
					<th>Monthly Payment</th>
					<th>APR</th>
				</tr>
			</thead>
			<tbody>
				{quotes && quotes.map((quote, i) => (
					<tr key={i}>
						<th>{quote.lenderName}</th>
						<th>{quote.loanType}</th>
						<th>{quote.interestRate}</th>
						<th>{quote.closingCosts}</th>
						<th>{quote.monthlyPayment}</th>
						<th>{quote.apr}</th>
					</tr>
				))}
			</tbody>
		</Table>
		{loading && (<Spinner animation="border" />)}
		{!loading && quotes.length == 0 &&(<h3>No quotes available</h3>)}
	</div>
)

export default QuotesTable