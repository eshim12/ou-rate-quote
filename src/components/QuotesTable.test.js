import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';

import QuotesTable from './QuotesTable';

describe('QuotesTable', () => {
	let component

	test('snapshot renders', () => {
		component = renderer.create(<QuotesTable quotes={[]}/>)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('Message appears when there are no quotes', () => {
		component = render(<QuotesTable quotes={[]}/>)
		expect(component.find("h3").text()).toEqual("No quotes available")
	})

	test('Snapshot renders with spinner', () => {
		component = renderer.create(<QuotesTable loading={true}/>)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('Message appears when there are no quotes', () => {
		component = render(<QuotesTable loading={true}/>)
		expect(component.find(".spinner-border").length).toBe(1)
	})

	test('Snapshot renders with data in table', () => {
		const quotes = [{lenderName: 'Lender', loanType: 'type', interestRate: '12.01', closingCosts: '2.01', monthlyPayment: '12', apr: '11.01'}]
		component = renderer.create(<QuotesTable quotes={quotes}/>)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('Data populates table', () => {
		const quotes = [{lenderName: 'Lender', loanType: 'type', interestRate: '12.01', closingCosts: '2.01', monthlyPayment: '12', apr: '11.01'}]
		component = render(<QuotesTable quotes={quotes}/>)
		expect(component.find('tbody').find('th').length).toBe(6)
	})
})