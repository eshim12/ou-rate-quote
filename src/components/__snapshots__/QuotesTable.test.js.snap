// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotesTable Snapshot renders with data in table 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <thead>
      <tr>
        <th>
          Lender
        </th>
        <th>
          Product
        </th>
        <th>
          Rate
        </th>
        <th>
          Closing Costs
        </th>
        <th>
          Monthly Payment
        </th>
        <th>
          APR
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          Lender
        </th>
        <th>
          type
        </th>
        <th>
          12.01
        </th>
        <th>
          2.01
        </th>
        <th>
          12
        </th>
        <th>
          11.01
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`QuotesTable Snapshot renders with spinner 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <thead>
      <tr>
        <th>
          Lender
        </th>
        <th>
          Product
        </th>
        <th>
          Rate
        </th>
        <th>
          Closing Costs
        </th>
        <th>
          Monthly Payment
        </th>
        <th>
          APR
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="spinner-border"
  />
</div>
`;

exports[`QuotesTable snapshot renders 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <thead>
      <tr>
        <th>
          Lender
        </th>
        <th>
          Product
        </th>
        <th>
          Rate
        </th>
        <th>
          Closing Costs
        </th>
        <th>
          Monthly Payment
        </th>
        <th>
          APR
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <h3>
    No quotes available
  </h3>
</div>
`;
